### PR TITLE
Fix two issues in ui/index.d.ts

### PR DESCRIPTION
**Pull request by 50Wliu**
This fixes two issues in the Typescript declarations for the ui package, specifically:
```
node_modules/vscode-azureextensionui/index.d.ts:1218:8 - error TS1040: 'async' modifier cannot be used in an ambient context.

1218 export async function createGenericClient(clientInfo?: ServiceClientCredentials | { credentials: ServiceClientCredentials; environment: Environment; }): Promise<ServiceClient>;
            ~~~~~

node_modules/vscode-azureextensionui/index.d.ts:1330:9 - error TS1086: An accessor cannot be declared in an ambient context.

1330     get onDidChangeFile(): Event<FileChangeEvent[]>;
             ~~~~~~~~~~~~~~~
```

I'm not completely confident about my change to `onDidChangeFile`, but that seems to be the way to do it according to the internet.

Also, is there any reason the declarations aren't auto-generated using the `--declaration` flag for tsc?

### CODE REVIEW

**Comment by nturinski** [comment]
Closing and reopening to kick-start build checks

**Comment by nturinski** [comment]
Thanks for your contribution!  I'm not actually running into any of these errors at all-- what version of Typescript are you running?

**Comment by 50Wliu** [comment]
Testing on a barebones project, I no longer see the second error, but a different version of the first one has popped up (TS 4.0.3).

Here's the simplest repro I could get:
index.ts
```ts
import { UserCancelledError } from 'vscode-azureextensionui'

throw new UserCancelledError();
```

package.json
```json
{
  "name": "test",
  "main": "index.js",
  "scripts": {
    "build": "tsc -p ./"
  },
  "dependencies": {
    "vscode-azureextensionui": "0.36.1"
  },
  "devDependencies": {
    "@types/vscode": "1.50.0",
    "typescript": "4.0.3"
  }
}
```

tsconfig.json
```json
{
  "compilerOptions": {
    "target": "es2019",
    "module": "commonjs"
  }
}
```

Errors:
```
node_modules/vscode-azureextensionui/index.d.ts:1190:16 - error TS1040: 'async' modifier cannot be used in an ambient context.

1190 export declare async function createExperimentationService(ctx: ExtensionContext, targetPopulation?: TargetPopulation): Promise<IExperimentationServiceAdapter>;
                    ~~~~~

node_modules/vscode-azureextensionui/index.d.ts:1262:8 - error TS1040: 'async' modifier cannot be used in an ambient context.

1262 export async function createGenericClient(clientInfo?: ServiceClientCredentials | { credentials: ServiceClientCredentials; environment: Environment; }): Promise<ServiceClient>;
            ~~~~~


Found 2 errors.
```

File: ui/index.d.ts [hunk 1]
```diff
@@ -1371,7 +1371,7 @@ export declare abstract class AzExtTreeFileSystem<TItem extends AzExtTreeItem> i
 
     public constructor(tree: AzExtTreeDataProvider);
 
-    public get onDidChangeFile(): Event<FileChangeEvent[]>;
+    public readonly onDidChangeFile: Event<FileChangeEvent[]>;
```
Comment (nturinski) [inline ui/index.d.ts:1374]: As you mentioned, I didn't get any errors with using get in our index.d.ts.  Could you change this back?

**Review by nturinski — COMMENTED** [review]
Removing the asyncs makes sense to me.  According to this: https://stackoverflow.com/questions/48448151/how-can-i-add-an-async-function-to-a-type-def-file, the async is really just for the implementation, not the contract itself.  As long as it returns a `Promise<T>`, we should be good.

**Comment by 50Wliu** [comment]
Done!

**Review by nturinski — APPROVED** [review]
Thanks a bunch for your contributions!

**Pull request merged**